### ISSUE TITLE
windows

### ISSUE THREAD

**Issue by ghost** [scaled/scaled#3]
How do I compile/run this on windows. Thanks.

**Comment by fpguy** [comment]
Would be nice if you added sbt support to the project and this way we'll be able to compile it anywhere.Thanks.

**Comment by jhigdon** [comment]
+1 fpguy. I even tried in cygwin and had a really bad time :(

**Comment by samskivert** [comment]
What's the preferred approach to deploying developer tools on Windows? If I make a standalone jar file that does everything that's needed, which could be run via "java -jar runscaled.jar" or something like that, would that be Windows-friendly?

How do you install and run SBT on Windows? Does it use a .bat file? I know on Unix/Mac SBT has a standalone jar file, and there's also the sbt-extras project which ships a shell script that launches SBT and does lots of smart stuff, but that's just as Windows-unfriendly as my shell script, so that seems like a bad example.

**Comment by OlegYch** [comment]
a single jar file would definitely work
sbt will work fine on windows using a single jar as well (http://www.scala-sbt.org/0.13/tutorial/Manual-Installation.html#Windows), though there is an installer available

**Comment by samskivert** [comment]
That's good to know. I was already leaning toward creating a single jar file bootstrap for spam anyway, because the shell script is fragile and relies on other tools being installed on the local machine, which is just more opportunities for failure.

I will still need the ability to checkout a Git repository. Does a Windows user usually install a command line executable version of Git or do you think I'll need to embed a Java Git client into the bootstrap jar?

**Comment by OlegYch** [comment]
i think it's fare to assume that git will be in path

**Comment by jhigdon** [comment]
I agree I think git can be assumed, it's pretty easy to get git on windows and many gui programs install their own git so there is often a cli version laying around somewhere. I installed SBT with the installer and it uses a bat that calls the sbt-launch.jar https://gist.github.com/jhigdon/ce5a608711c5d9745651
